htlcswitch/switch.go

```diff
@@ -704,12 +708,19 @@ func (s *Switch) ForwardPackets(linkQuit <-chan struct{},
 	select {
 	case <-linkQuit:
 		return nil
-	case <-s.quit:
+
+	case <-s.gm.Done():
 		return nil
+
 	default:
-		// Spawn a goroutine to log the errors returned from failed packets.
-		s.wg.Add(1)
-		go s.logFwdErrs(&numSent, &wg, fwdChan)
+		// Spawn a goroutine to log the errors returned from failed
+		// packets.
+		ok := s.gm.Go(context.TODO(), func(ctx context.Context) {
+			s.logFwdErrs(ctx, &numSent, &wg, fwdChan)
+		})
+		if !ok {
```

Comment:
same here - why do we need this check? I think `s.logFwdErrs` will listen on `<-s.gm.Done():` and quit?

Reply:
This is a similar situation. We should handle it the same way if the goroutine manager is stopped before the Go method is executed.